mule-module-apikit/src/test/java/org/mule/module/apikit/UrlUtilsTestCase.java

```diff
@@ -0,0 +1,339 @@
+package org.mule.module.apikit;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+public class UrlUtilsTestCase
+{
+    @Test
+    public void replaceRelativeBaseUriWithFull()
+    {
+        String newRaml = UrlUtils.replaceBaseUri(ramlRelativeUri, "http://localhost:8081/api");
+        assertTrue(newRaml.contains("baseUri: http://localhost:8081/api"));
+    }
+
+    @Test
+    public void replaceRelativeBaseUriWithRelative()
+    {
+        String newRaml = UrlUtils.replaceBaseUri(ramlRelativeUri, "/newapi");
+        assertTrue(newRaml.contains("baseUri: /newapi"));
+    }
+
+    @Test
+    public void replaceFullBaseUriWithFull()
+    {
+        String newRaml = UrlUtils.replaceBaseUri(ramlFullUri, "http://localhost:8081/api");
+        assertTrue(newRaml.contains("baseUri: http://localhost:8081/api"));
+    }
+
+    @Test
+    public void replaceFullBaseUriWithRelative()
+    {
+        String newRaml = UrlUtils.replaceBaseUri(ramlFullUri, "/newapi");
+        assertTrue(newRaml.contains("baseUri: /newapi"));
+    }
+
+    String ramlRelativeUri = "#%RAML 1.0\n" +
```

Comment:
you only need a raml with a baseUri for the test, the following should be enough:
```
#%RAML 1.0
title: Admin
baseUri: /api
```